lib/subordinateio.c

```diff
@@ -745,6 +747,33 @@ gid_t sub_gid_find_free_range(gid_t min, gid_t max, unsigned long count)
 	return start == ULONG_MAX ? (gid_t) -1 : start;
 }
 
+static bool get_owner_id(const char *owner, enum subid_type id_type, char *id)
+{
+	struct passwd *pw;
+	struct group *gr;
+
+	switch (id_type) {
+	case ID_TYPE_UID:
+		pw = getpwnam(owner);
+		if (pw == NULL) {
+			return false;
+		}
+		snprintf(id, ID_SIZE, "%u", pw->pw_uid);
```

Comment:
Sorry could you change both snprintfs to checking the return value as is done e..g here https://github.com/lxc/lxc/blob/master/src/lxc/lsm/apparmor.c#L406 ?

There are various schools of thoughts on string printing, but when I see *that* then I know every corner case is handled and I don't have to think it through.